search/includes/classes/class-search.php

```diff
@@ -1037,6 +1040,67 @@ public function filter__ep_do_intercept_request( $request, $query, $args, $failu
 		return $response;
 	}
 
+	/**
+	 * Mirror write requests to migration ES hosts during migration.
+	 *
+	 * @param array $query The original query array.
+	 * @param array $args The original request args.
+	 * @param string|null $type The type of request.
+	 */
+	protected function mirror_write_to_migration_hosts( $query, $args, $type = null ) {
+		if (
+			! defined( 'VIP_ELASTICSEARCH_VERSION' ) ||
+			constant( 'VIP_ELASTICSEARCH_VERSION' ) !== 'migration_in_progress' ||
+			! defined( 'VIP_ELASTICSEARCH_MIGRATION_ENDPOINTS' ) ||
+			! is_array( constant( 'VIP_ELASTICSEARCH_MIGRATION_ENDPOINTS' ) )
+		) {
+			return;
+		}
+
+		$allowed_write_types = [
+			'index',
+			'bulk_index',
+			'delete',
+		];
+		if ( ! $type || ! in_array( $type, $allowed_write_types, true ) ) {
+			return;
+		}
+
+		$migration_hosts = constant( 'VIP_ELASTICSEARCH_MIGRATION_ENDPOINTS' );
+		if ( ! is_array( $migration_hosts ) || empty( $migration_hosts ) ) {
+			return;
+		}
+
+		foreach ( $migration_hosts as $migration_host ) {
+			$migration_url = preg_replace(
```

Comment:
I think this needs a random pick vs iterating over each host, cause that's a request per each one.